### PR TITLE
Serialize Uri for no input shape

### PR DESCRIPTION
**Pull request by kyleknap**
There was an issue where if the operation had no input shape, we would not use the modeled URI and just use ``'/'`` because that is the default.

cc @jamesls @mtdowling @rayluo

### CODE REVIEW

File: tests/unit/protocols/input/rest-xml.json [hunk 1]
```diff
@@ -73,16 +73,16 @@
       {
         "given": {
           "http": {
-            "method": "PUT",
+            "method": "GET",
```
Comment (jamesls) [inline tests/unit/protocols/input/rest-xml.json]: Why change this instead of adding a new test?  If the PUT case no longer valid if you send an empty body?

Comment (kyleknap) [inline tests/unit/protocols/input/rest-xml.json]: @jamesls 
I was the one to originally added this test case. I added it to help improve coverage on our serializers. The PUT method was just used because of me just copying and pasting from previous examples. The test case was to cover if there was no input shape for the operation and so I believe the expected value is incorrect so it needed changing. I also cannot think of an example of a ``PUT`` request with no input parameters.

**Comment by jamesls** [comment]
:shipit:  Looks good.
